### ISSUE TITLE
Backend memory leak

### ISSUE THREAD

**Issue by jordanh** [ParabolInc/action#1196]
## Issue - Bug

Looking at the historical data on datadog, we probably have a backend memory leak. Check out this graph:

![image](https://user-images.githubusercontent.com/103000/28895158-29b7fbae-778c-11e7-9748-a1b8680a647b.png)

This is the rss for the `web.web.1` container on production. You can see utilization go down after a deploy, and the increase to fill the container over time.

- **OS:** OSX
- **Browser:** Chrome

**Comment by jordanh** [comment]
I took a little time this evening to look into this. The first step was to add a little bit of tooling. After some reading I decided to give https://github.com/marcominetti/node-memwatch a try. I've been playing with it on this branch: https://github.com/ParabolInc/action/tree/memleak-diag

It looks like there are Array objects *somewhere* that are growing unbounded. It's not exactly clear what these objects are. We'll probably need to take some heapdumps and see if anything looks familiar (a la https://www.alexkras.com/simple-guide-to-finding-a-javascript-memory-leak-in-node-js/).

@mattkrick you want to have a memory leak party at your house next week?? This would be a blast to do on Tuesday.
